src/classes/CON_PrimaryAffls_TDTM.cls

```diff
@@ -84,18 +84,22 @@ public class CON_PrimaryAffls_TDTM extends TDTM_Runnable {
             TDTM_ProcessControl.setRecursionFlag(TDTM_ProcessControl.registeredTrigger.CON_PrimaryAffls_TDTM_After_Insert, true);
         }
 
-        // BEFORE UPDATE
-        if ( newlist != null && triggerAction == TDTM_Runnable.Action.AfterUpdate && !TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.CON_PrimaryAffls_TDTM_After_Update)) {
-            for (integer i=0; i<newlist.size(); i++) {
+        // AFTER UPDATE
+        // Do not run if the reentrant flag is set OR if the contacts are being updated as a result of the AFFL_AccRecordType_TDTM trigger, as
+        // that TDTM class already handles the Affiliation record updates.
+        if (newlist != null && triggerAction == TDTM_Runnable.Action.AfterUpdate 
+                && !TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.CON_PrimaryAffls_TDTM_After_Update)
+                && !TDTM_ProcessControl.getRecursionFlag(TDTM_ProcessControl.registeredTrigger.AFFL_AccRecordType_TDTM)) {
```

Comment:
@skosgi1 @lifewithryan I think this might be the better fix - still not sure if it's the "correct" fix but I think it's safe to ignore the logic here if we're in the middle of updating contacts & affiliation records as a result of an account record type change, since the updates are all handled in that class instead.

Reply:
We just have to watch out for side effects here that we may not be aware of, but seems like it'd be okay.